lib/src/stream_subscription_transformer.dart

```diff
@@ -18,10 +18,14 @@ typedef void _SimpleHandler<T>(StreamSubscription<T> inner);
 /// [StreamSubscription.resume] is called, the corresponding handler is invoked.
 /// By default, handlers just forward to the underlying subscription.
 ///
-/// This guarantees that once the transformed [StreamSubscription] has been
-/// canceled, no further callbacks will be invoked. The [handlePause] and
-/// [handleResume] are invoked regardless of whether the subscription is paused
-/// already or not.
+/// Guarantees that once the transformed [StreamSubscription] has been canceled,
+/// no further callbacks will be invoked. The [handlePause] and [handleResume]
```

Comment:
But, as I read it, that isn't guaranteed.

If the `onCancel` handler does nothing, you can get further event callbacks after calling `cancel` on the outer subscription. The only way to avoid that with the current code is for the `onCancel` to stop events from the inner subscription (either by cancel or pause). Removing the callbacks is one option, then at least you don't get callbacks, but since you can set them again, you also need to modify the `onData`, `onError` and `onDone` methods to not forward after a cancel.

For `pause` it's even worse because you can't remove the event handlers. 

So you, and this class, don't guarantee anything here. It is a requirement on the `handlePause` and `handleCancel` that they stop events.